src/query/QueryManager.jl

```diff
@@ -2,36 +2,52 @@
 
 function match(transaction::AbstractCoreTransaction, query::String, options = Proto.Options())
     db_result =  stream(transaction, QueryManagerRequestBuilder.match_req(query, options))
-    maps = map(x->ConceptMap(x), [entry.query_manager_res_part.match_res_part.answers for entry in db_result])
-    result = reduce(vcat, maps)
+    if db_result !== nothing
+        maps = map(x->ConceptMap(x), [entry.query_manager_res_part.match_res_part.answers for entry in db_result])
+        result = reduce(vcat, maps)
+    else
+        result = nothing
+    end
     return result
 end
 
 function match_aggregate(transaction::AbstractCoreTransaction, query::String, options = Proto.Options())
     db_result =  execute(transaction, QueryManagerRequestBuilder.match_aggregate_req(query, options))
-    result = [_read_proto_number(entry.query_manager_res.match_aggregate_res.answer) for entry in db_result]
+    result = _read_proto_number(db_result.query_manager_res.match_aggregate_res.answer)
     return result
 end
 
 function match_group(transaction::AbstractCoreTransaction, query::String, options = Proto.Options())
     db_result =  stream(transaction, QueryManagerRequestBuilder.match_group_req(query, options))
-    maps =  [ConceptMapGroup(item.query_manager_res_part.match_group_res_part.answers) for item in db_result]
-    result = reduce(vcat, maps)
+    if db_result !== nothing
+        maps =  [ConceptMapGroup(item.query_manager_res_part.match_group_res_part.answers) for item in db_result]
+        result = reduce(vcat, maps)
+    else
+        result = nothing
+    end
     return result
 end
 
 function match_group_aggregate(transaction::AbstractCoreTransaction, query::String, options = Proto.Options())
     db_result =  stream(transaction, QueryManagerRequestBuilder.match_group_aggregate_req(query, options))
-    maps = [NumericGroup(item.match_group_aggregate_res_part.answers) for item in db_result]
-    result = reduce(vcat, maps)
+    if db_result !== nothing
+        maps = [NumericGroup(item.match_group_aggregate_res_part.answers) for item in db_result]
+        result = reduce(vcat, maps)
+    else
+        result = nothing
+    end
     return result
 end
 
 function insert(transaction::AbstractCoreTransaction, query::String, options = Proto.Options())
     db_result = stream(transaction, QueryManagerRequestBuilder.insert_req(query, options))
-    answers = [entry.query_manager_res_part.insert_res_part.answers for entry in db_result]
-    maps = map(ConceptMap, answers)
-    result = reduce(vcat, maps)
+    if db_result !== nothing
+        answers = [entry.query_manager_res_part.insert_res_part.answers for entry in db_result]
+        maps = map(ConceptMap, answers)
```

Comment:
Can't `ConceptMap` be applied in the line above, instead of having to create another temporary vector?

Reply:
Yes, of course. I chose this form for a better readability and a better debugging. For now bringing all together in one line it is kinda reading overkill :-). For now I would let it as it is